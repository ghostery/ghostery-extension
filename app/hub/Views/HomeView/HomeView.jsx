/**
 * Home View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file.
 */

import React from 'react';

/**
 * @class Implement the Home View for the Ghostery Hub
 * @memberof HubComponents
 */
const HomeView = () => (
	<div>
		<div className="row ready-header">
			<div className="columns small-6">
				<div className="ghosty-dialogue" />
			</div>
			<div className="columns small-6">
				<h1>Ghostery is Ready!</h1>
				<p>you are now protected with the Ghostery recommended default settings.</p>
				<span className="bold">Start browsing!</span>
				<label>
					<input type="checkbox" />
					Support Ghostery by sharing Human Web & Analytics data.
					<a>Learn more.</a>
				</label>
			</div>
		</div>
		<div className="row optimize-create">
			<div className="columns small-6">
				<span>Optimze your ghostery experience</span>
			</div>
			<div className="columns small-6">
				<a>Create Account</a>
			</div>
		</div>
		<div className="row tutorial-custom">
			<div className="columns small-6">
				tutorial
			</div>
			<div className="columns small-6">
				customize setup
			</div>
		</div>
		<div className="row supporter">
			<div className="columns small-6">
				become a ghostery support text
			</div>
			<div className="columns small-6">
				support button
			</div>
		</div>
	</div>
);


export default HomeView;
