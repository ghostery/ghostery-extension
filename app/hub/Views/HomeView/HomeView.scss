/**
 * Home View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Home View
.HomeView {
	padding-top: 45px;
	padding-bottom: 25px;
	color: #4a4a4a;
}
.HomeView--bolded {
	font-weight: 700;
}
.HomeView--pad-left {
	padding-left: 12px;
}
.HomeView__header img {
	max-width: 156px;
	margin-right: 23px;
}
.HomeView__header h1 {
	color: #4a4a4a;
	margin-bottom: 19px;
}
.HomeView__headerTagline {
	font-size: 20px;
	line-height: 38px;
	color: #4a4a4a;
}
.HomeView__supportContainer {
	margin-top: 12px;
	margin-left: -12px;
}
.HomeView__subHeader {
	margin-top: 12px;
	font-size: 16px;
	font-weight: 500;
	line-height: 32px;
}
.HomeView__subHeader a {
	text-decoration: underline;
}
.HomeView__onboarding {
	margin-top: 7px;
	padding: 27px 12px 30px;
	border-radius: 4px;
	background-color: #f0f3f4;
}
.HomeView__onboardingFeature {
	margin: 0 8px;
	padding: 0 36px;
}
.HomeView__onboardingFeatureDivider {
	margin: 7px 0 4px;
	width: 1px;
	padding: 0;
	background-color: #979797;
}
.HomeView__supporter {
	margin-top: 22px;
	padding: 22px 12px;
	background-color: #f0f3f4;
	border-radius: 4px;
}
.HomeView__supporterFeature {
	padding: 0 36px;
}
.HomeView__featureIcon {
	height: 50px;
	width: 50px;
	margin: 0 24px;
	background-repeat: no-repeat;
	background-position: center center;
}
.HomeView__onboardingFeature.feature-tutorial .HomeView__featureIcon {
	background-image: buildIconWand(#4a4a4a, #daf4ff);
}
.HomeView__onboardingFeature.feature-tutorial-complete .HomeView__featureIcon {
	width: 100px;
	margin: 0 0 0 50px;
	background-image: buildIconWandCheck(#4a4a4a, #daf4ff, #1dafed);
}
.HomeView__onboardingFeature.feature-setup .HomeView__featureIcon {
	background-image: buildIconClipboard(#4a4a4a, #daf4ff);
}
.HomeView__onboardingFeature.feature-setup-complete .HomeView__featureIcon {
	width: 100px;
	margin: 0 0 0 50px;
	background-image: buildIconClipboardCheck(#4a4a4a, #daf4ff, #1dafed);
}
.HomeView__featureIcon.feature-supporter {
	background-image: buildIconFlag(#4a4a4a, #daf4ff);
}
.HomeView__featureTitle {
	text-align: center;
	font-size: 18px;
	padding: 22px 0 10px;
}
.HomeView__featureText {
	text-align: center;
	font-size: 14px;
	line-height: 27px;
}
.HomeView__featureButton {
	margin: 0;
}
.HomeView__onboarding .HomeView__featureButton {
	margin-top: 26px;
	min-width: 130px;
}
