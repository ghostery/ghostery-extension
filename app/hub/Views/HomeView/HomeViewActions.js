/**
 * Home View Action creators
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { log, sendMessageInPromise } from '../../utils';
import { GET_HOME_PROPS, SET_METRICS } from './HomeViewConstants';

export function getHomeProps() {
	return function (dispatch) {
		return sendMessageInPromise(GET_HOME_PROPS).then((data) => {
			dispatch({
				type: GET_HOME_PROPS,
				data,
			});
		}).catch((err) => {
			log('homeView Action getHomeProps Error', err);
		});
	};
}

export function setMetrics(actionData) {
	return function (dispatch) {
		return sendMessageInPromise(SET_METRICS, actionData).then((data) => {
			dispatch({
				type: SET_METRICS,
				data,
			});
		}).catch((err) => {
			log('homeView Action setMetrics Error', err);
		});
	};
}
