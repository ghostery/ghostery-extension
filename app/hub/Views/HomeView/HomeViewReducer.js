/**
 * Reducer used in the Home View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { GET_HOME_PROPS, SET_METRICS } from './HomeViewConstants';

const initialState = {};

function HomeViewReducer(state = initialState, action) {
	switch (action.type) {
		case GET_HOME_PROPS: {
			const {
				setup_complete,
				tutorial_complete,
				enable_metrics,
				account_text,
				account_link,
			} = action.data;
			return Object.assign({}, state, {
				home: Object.assign({}, state.home, {
					setup_complete,
					tutorial_complete,
					enable_metrics,
					account_text,
					account_link,
				}),
			});
		}
		case SET_METRICS: {
			const { enable_metrics } = action.data;
			return Object.assign({}, state, {
				home: Object.assign({}, state.home, { enable_metrics }),
			});
		}

		default: return state;
	}
}

export default HomeViewReducer;
