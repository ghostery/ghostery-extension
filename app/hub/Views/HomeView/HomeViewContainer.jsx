/**
 * Tutorial View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import QueryString from 'query-string';
import HomeView from './HomeView';

/**
 * @class Implement the Home View Container for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class HomeViewContainer extends React.Component {
	constructor(props) {
		super(props);

		const { justInstalled } = QueryString.parse(window.location.search);
		this.state = {
			justInstalled: justInstalled === 'true',
		};

		const title = t('hub_home_page_title');
		window.document.title = title;

		this.props.actions.getHomeProps();
	}

	/**
	* Function to handle toggling Metrics Opt-In
	*/
	_handleToggleMetrics = () => {
		const enable_metrics = !this.props.home.enable_metrics;
		this.props.actions.setMetrics({ enable_metrics });
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Home View of the Hub app
	 */
	render() {
		// ToDo: Get these from action, reducer and props. Will be on this.props.home
		// These are passed as props so we can the user's email and link to their account when they are signed in
		const account_text = t('hub_home_subheader_create_account');
		const account_link = '/create-account';

		const { justInstalled } = this.state;
		const {
			setup_complete,
			tutorial_complete,
			enable_metrics,
		} = this.props.home;
		const childProps = {
			justInstalled,
			setup_complete,
			tutorial_complete,
			enable_metrics,
			changeMetrics: this._handleToggleMetrics,
			account_text,
			account_link,
		};

		return <HomeView {...childProps} />;
	}
}

// PropTypes ensure we pass required props of the correct type
// Note: isRequired is not needed when a prop has a default value
HomeViewContainer.propTypes = {
	home: PropTypes.shape({
		setup_complete: PropTypes.bool,
		tutorial_complete: PropTypes.bool,
		enable_metrics: PropTypes.bool,
		account_text: PropTypes.string,
		account_link: PropTypes.string,
	}),
};

// Default props used throughout the Setup flow
HomeViewContainer.defaultProps = {
	home: {
		setup_complete: false,
		tutorial_complete: false,
		enable_metrics: false,
		account_text: '',
		account_link: '',
	},
};

export default HomeViewContainer;
