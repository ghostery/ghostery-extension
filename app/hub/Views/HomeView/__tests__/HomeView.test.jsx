/**
 * Home View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import HomeView from '../HomeView';

describe('app/hub/Views/HomeView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('home view is rendered correctly: all true', () => {
			const initialState = {
				justInstalled: true,
				setup_complete: true,
				tutorial_complete: true,
				enable_metrics: true,
				changeMetrics: () => {},
				account_text: 'test create account',
				account_link: '/test/create-account',
			};

			const component = renderer.create(
				<MemoryRouter>
					<HomeView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('home view is rendered correctly: all false', () => {
			const initialState = {
				justInstalled: false,
				setup_complete: false,
				tutorial_complete: false,
				enable_metrics: false,
				changeMetrics: () => {},
				account_text: 'test@example.com',
				account_link: '/test/user-profile',
			};

			const component = renderer.create(
				<MemoryRouter>
					<HomeView {...initialState} />
				</MemoryRouter>			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				justInstalled: true,
				setup_complete: false,
				tutorial_complete: false,
				enable_metrics: false,
				changeMetrics: jest.fn(),
				account_text: 'test username',
				account_link: '/test/url',
			};

			const component = shallow(<HomeView {...initialState} />);
			expect(component.find('.HomeView').length).toBe(1);
			expect(component.find('.button').length).toBe(3);
			expect(component.find('.HomeView__header').length).toBe(1);
			expect(component.find('.HomeView__subHeader').length).toBe(1);
			expect(component.find('.HomeView__supportContainer .clickable').length).toBe(1);

			expect(initialState.changeMetrics.mock.calls.length).toBe(0);
			component.find('.HomeView__supportContainer .clickable').simulate('click');
			expect(initialState.changeMetrics.mock.calls.length).toBe(1);

			expect(component.find('.HomeView__headerTagline').length).toBe(2);
			component.setProps({ justInstalled: false });
			expect(component.find('.HomeView__headerTagline').length).toBe(1);

			expect(component.find('.HomeView__featureButton.hollow').length).toBe(0);
			component.setProps({ tutorial_complete: true });
			expect(component.find('.HomeView__featureButton.hollow').length).toBe(1);

			component.setProps({ setup_complete: true });
			expect(component.find('.HomeView__featureButton.hollow').length).toBe(2);
		});
	});
});
