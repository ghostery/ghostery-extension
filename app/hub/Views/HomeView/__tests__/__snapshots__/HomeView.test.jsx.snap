// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/HomeView component Snapshot tests with react-test-renderer home view is rendered correctly: all false 1`] = `
<div
  className="HomeView row align-center"
>
  <div
    className="columns small-12 medium-10 large-8"
  >
    <div
      className="HomeView__header row align-center-middle"
    >
      <img
        className="columns"
        src="/app/images/hub/home/ghosty-bubble-heart.svg"
      />
      <div
        className="HomeView__title columns"
      >
        <h1>
          hub_home_header_text
        </h1>
        <div
          className="HomeView__headerTagline HomeView--bolded"
        >
          hub_home_header_tagline_2
        </div>
        <div
          className="HomeView__supportContainer HomeView--pad-left"
        >
          <span>
            hub_home_header_info
          </span>
          <a
            alt="hub_home_header_info_link"
            href="https://www.ghostery.com/faqs/"
            rel="noopener noreferrer"
            target="_blank"
          >
            hub_home_header_info_link
          </a>
        </div>
        <div
          className="HomeView__supportContainer flex-container align-middle"
        >
          <div
            className="ToggleCheckbox"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </div>
          <span
            className="clickable"
            onClick={[Function]}
          >
            hub_home_header_checkbox_label
          </span>
        </div>
      </div>
    </div>
    <div
      className="HomeView__subHeader row align-justify"
    >
      <div>
        hub_home_subheader_optimize
      </div>
      <a
        aria-current={null}
        className={undefined}
        href="/test/user-profile"
        onClick={[Function]}
        style={undefined}
      >
        test@example.com
      </a>
    </div>
    <div
      className="HomeView__onboarding row"
    >
      <div
        className="HomeView__onboardingFeature columns flex-container align-middle flex-dir-column feature-tutorial"
      >
        <div
          className="HomeView__featureIcon"
        />
        <div
          className="HomeView__featureTitle"
        >
          hub_home_feature_tutorial_title
        </div>
        <div
          className="HomeView__featureText flex-child-grow"
        >
          hub_home_feature_tutorial_text
        </div>
        <a
          aria-current={null}
          className="HomeView__featureButton button primary"
          href="/tutorial"
          onClick={[Function]}
          style={undefined}
        >
          hub_home_feature_tutorial_button
        </a>
      </div>
      <div
        className="HomeView__onboardingFeatureDivider column shrink"
      />
      <div
        className="HomeView__onboardingFeature columns flex-container align-middle flex-dir-column feature-setup"
      >
        <div
          className="HomeView__featureIcon"
        />
        <div
          className="HomeView__featureTitle"
        >
          hub_home_feature_setup_title
        </div>
        <div
          className="HomeView__featureText flex-child-grow"
        >
          hub_home_feature_setup_text
        </div>
        <a
          aria-current={null}
          className="HomeView__featureButton button primary"
          href="/setup"
          onClick={[Function]}
          style={undefined}
        >
          hub_home_feature_setup_button
        </a>
      </div>
    </div>
    <div
      className="HomeView__supporter row"
    >
      <div
        className="HomeView__supporterFeature columns flex-container align-middle"
      >
        <div
          className="HomeView__featureText columns"
        >
          hub_home_feature_supporter_text
        </div>
        <div
          className="HomeView__featureIcon columns shrink feature-supporter"
        />
        <div
          className="columns flex-container align-center-middle"
        >
          <a
            aria-current={null}
            className="HomeView__featureButton button primary"
            href="/supporter"
            onClick={[Function]}
            style={undefined}
          >
            hub_home_feature_supporter_button
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app/hub/Views/HomeView component Snapshot tests with react-test-renderer home view is rendered correctly: all true 1`] = `
<div
  className="HomeView row align-center"
>
  <div
    className="columns small-12 medium-10 large-8"
  >
    <div
      className="HomeView__header row align-center-middle"
    >
      <img
        className="columns"
        src="/app/images/hub/home/ghosty-bubble-heart.svg"
      />
      <div
        className="HomeView__title columns"
      >
        <h1>
          hub_home_header_text
        </h1>
        <div
          className="HomeView__headerTagline"
        >
          hub_home_header_tagline
        </div>
        <div
          className="HomeView__headerTagline HomeView--bolded"
        >
          hub_home_header_tagline_2
        </div>
        <div
          className="HomeView__supportContainer HomeView--pad-left"
        >
          <span>
            hub_home_header_info
          </span>
          <a
            alt="hub_home_header_info_link"
            href="https://www.ghostery.com/faqs/"
            rel="noopener noreferrer"
            target="_blank"
          >
            hub_home_header_info_link
          </a>
        </div>
        <div
          className="HomeView__supportContainer flex-container align-middle"
        >
          <div
            className="ToggleCheckbox ToggleCheckbox--active"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </div>
          <span
            className="clickable"
            onClick={[Function]}
          >
            hub_home_header_checkbox_label
          </span>
        </div>
      </div>
    </div>
    <div
      className="HomeView__subHeader row align-justify"
    >
      <div>
        hub_home_subheader_optimize
      </div>
      <a
        aria-current={null}
        className={undefined}
        href="/test/create-account"
        onClick={[Function]}
        style={undefined}
      >
        test create account
      </a>
    </div>
    <div
      className="HomeView__onboarding row"
    >
      <div
        className="HomeView__onboardingFeature columns flex-container align-middle flex-dir-column feature-tutorial-complete"
      >
        <div
          className="HomeView__featureIcon"
        />
        <div
          className="HomeView__featureTitle"
        >
          hub_home_feature_tutorial_title
        </div>
        <div
          className="HomeView__featureText flex-child-grow"
        >
          hub_home_feature_tutorial_text
        </div>
        <a
          aria-current={null}
          className="HomeView__featureButton button primary hollow"
          href="/tutorial"
          onClick={[Function]}
          style={undefined}
        >
          hub_home_feature_tutorial_button_alt
        </a>
      </div>
      <div
        className="HomeView__onboardingFeatureDivider column shrink"
      />
      <div
        className="HomeView__onboardingFeature columns flex-container align-middle flex-dir-column feature-setup-complete"
      >
        <div
          className="HomeView__featureIcon"
        />
        <div
          className="HomeView__featureTitle"
        >
          hub_home_feature_setup_title
        </div>
        <div
          className="HomeView__featureText flex-child-grow"
        >
          hub_home_feature_setup_text
        </div>
        <a
          aria-current={null}
          className="HomeView__featureButton button primary hollow"
          href="/setup"
          onClick={[Function]}
          style={undefined}
        >
          hub_home_feature_setup_button_alt
        </a>
      </div>
    </div>
    <div
      className="HomeView__supporter row"
    >
      <div
        className="HomeView__supporterFeature columns flex-container align-middle"
      >
        <div
          className="HomeView__featureText columns"
        >
          hub_home_feature_supporter_text
        </div>
        <div
          className="HomeView__featureIcon columns shrink feature-supporter"
        />
        <div
          className="columns flex-container align-center-middle"
        >
          <a
            aria-current={null}
            className="HomeView__featureButton button primary"
            href="/supporter"
            onClick={[Function]}
            style={undefined}
          >
            hub_home_feature_supporter_button
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
